includes/hooks/template.php

```diff
@@ -82,11 +82,9 @@
 add_action( 'sensei_single_course_content_inside_after', array( 'Sensei_Course', 'the_course_lessons_title' ), 9 );
 
 // @since 1.9.0
-// hooks in the course lessons query and remove it at the end
-// also loading the course lessons template in the middle
-add_action( 'sensei_single_course_lessons_before', array( 'Sensei_Course', 'load_single_course_lessons_query' ) );
+// @since 3.0.0 Use secondary query to loop the lessons.
+// Load the course lessons template
 add_action( 'sensei_single_course_content_inside_after', 'course_single_lessons', 10 );
-add_action( 'sensei_single_course_lessons_after', array( 'Sensei_Utils', 'restore_wp_query' ) );
```

Comment:
With the new approach we should call `wp_reset_postdata` to make sure that global post data are restored after the custom loop is completed. This should be better done in `lessons.php` though and not with an action.

Reply:
It's done in this line, after to get the template (an run the query):
https://github.com/Automattic/sensei/pull/3032/files#diff-978209e01c6c9e4491c170874c427461R31

If you think here is not a good place to do it, I accept suggestions :)

Reply:
Oups sorry I missed that.